Add V1 API Version

wallrj opened the pull request:
* Copied v1beta1 API files to v1
* Updated package names
* Added v1 API to the code generation configuration
* Ran update-all.sh
* Copied v1beta1 CRD version definitions to v1
* Updated CRDs to make v1, the storage version
* Updated filter-crds to only retain the v1 version
* Updated conversion integration tests

Part of: https://github.com/jetstack/cert-manager/issues/3156

```release-note
Add `v1` API version
```

meyskens commented on pkg/apis/certmanager/v1/types.go:
```diff
@@ -0,0 +1,193 @@
+// Deprecated annotation names for Secrets
+// These will be removed in a future release.
+const (
+	DeprecatedIssuerNameAnnotationKey = "certmanager.k8s.io/issuer-name"
```
can be removed

wallrj replied:
I think I'll do that in a small followup PR, along with any other changes that we want to make to the v1 API before release.
I intended to keep this PR as mechanical as possible and as a record of the changes needed to add a new API version.
Added as an item to https://github.com/jetstack/cert-manager/issues/3156

wallrj replied:
Ok. I'm doing this here after all, since we're also making changes to the API field names in this PR.

meyskens commented:
Seems the YAML is missing, not sure if that was on purpose or not?
You should also
- Make it the stored version
- Modify /hack/filter-crd to only release the v1 API

wallrj commented:
Thanks for the review @meyskens. I've addressed / answered your comments. Is there anything I've missed? 
I've tried to keep the commits quite self contained.

meyskens commented:
@wallrj looking good to me after the tests pass, we should also have a quick think about the API if we want to change something before it's too late

wallrj commented:
Rebased on latest master and updated the conversion integration tests accordingly.

I have not updated the cmctl convert tool or its integration tests. 
I'd rather do that in a separate PR.

wallrj commented:
/test pull-cert-manager-e2e-v1-19
/test pull-cert-manager-e2e-v1-17
/test pull-cert-manager-e2e-v1-16
/test pull-cert-manager-e2e-v1-15
/test pull-cert-manager-e2e-v1-14
/test pull-cert-manager-e2e-v1-13
/test pull-cert-manager-e2e-v1-12
/test pull-cert-manager-e2e-v1-11

wallrj commented:
/test pull-cert-manager-e2e-v1-19
/test pull-cert-manager-e2e-v1-17
/test pull-cert-manager-e2e-v1-16
/test pull-cert-manager-e2e-v1-15
/test pull-cert-manager-e2e-v1-14
/test pull-cert-manager-e2e-v1-13
/test pull-cert-manager-e2e-v1-12
/test pull-cert-manager-e2e-v1-11

wallrj commented:
Changing storage version and using v1 API in legacy CRD can't work until we update all the controllers and E2E tests scripts to use the v1 API, so I removed those commits from this PR.

We agreed to make those changes in a followup PR at the same time as those controller changes.

wallrj commented:
/test pull-cert-manager-e2e-v1-19
/test pull-cert-manager-e2e-v1-17
/test pull-cert-manager-e2e-v1-16
/test pull-cert-manager-e2e-v1-15
/test pull-cert-manager-e2e-v1-14
/test pull-cert-manager-e2e-v1-13
/test pull-cert-manager-e2e-v1-12
/test pull-cert-manager-e2e-v1-11

meyskens commented:
/kind feature

meyskens commented on pkg/apis/certmanager/v1/types.go:
```diff
@@ -47,13 +47,6 @@ const (
-// Deprecated annotation names for Secrets
-// These will be removed in a future release.
-const (
-	DeprecatedIssuerNameAnnotationKey = "certmanager.k8s.io/issuer-name"
```
Needs removing from being used also: https://github.com/jetstack/cert-manager/blob/497e64de4b6920c3b967aff855e1fb2af2b463b4/pkg/controller/certificates/internal/secretsmanager/secret.go#L212

meyskens commented:
/test pull-cert-manager-e2e-v1-19
/test pull-cert-manager-e2e-v1-17
/test pull-cert-manager-e2e-v1-16
/test pull-cert-manager-e2e-v1-15
/test pull-cert-manager-e2e-v1-14
/test pull-cert-manager-e2e-v1-13
/test pull-cert-manager-e2e-v1-12
/test pull-cert-manager-e2e-v1-11

meyskens commented:
/approve
/lgtm
